### ISSUE TITLE
Maestral 1.4.6 unable to open

### ISSUE THREAD

**Issue by xiamenese** [SamSchott/maestral#416]
**Describe the bug**
Maestral 1.4.6 opened on installation. I then rebooted the computer—as v. 145 had problems following reboot—and v. 1.4.6 also fails to open. Failure message: "Could not access sys._trackback".

**To Reproduce**

Install Maestral 1.4.6; reboot computer.

**Expected behaviour**

Maestral running as before reboot.
**System:**

 - Maestral version: 1.4.6
 - Python version: 
 - OS: MacOS 11.4
 - MacOS BigSur
 - PyQt version (for Linux GUI): N/A

**Additional context**

2015 5K iMac, Intel Core i5, 16GB RAM, Internal SATA SSD.

**Comment by SamSchott** [comment]
Oh, that is strange. Are you using the app bundle or an installation from PyPI?

**Comment by SamSchott** [comment]
Also, just to clarify, are you seeing this issue only on reboot? Or also when manually launching Maestral?

**Comment by xiamenese** [comment]
It's happening on boot-up, but after two or three attempts yesterday to 
open it manually it worked. This morning when I booted the iMac it 
failed with the same message, but then opened manually.

I installed using the DMG, not from PyPI.

:)

Mark

SamSchott wrote:

**Comment by philippgerard** [comment]
Same here after updating to 1.4.6 on boot up.

> Could not access sys._traceback

Clicking "ok" seems to not prevent Maestral from working, though. The app is still active in the toolbar and reports to work just fine when clicking on it.

**Comment by SamSchott** [comment]
That is all very strange. The error message regarding accessing `sys._traceback` appears to be just a symptom of another error during startup. I have fixed the handling now to show an actual traceback instead of this obscure message but I am still confused to what the underlying issue is.

Is there anything useful in the log from those incidences? You can view them with the CLI command `maestral log show -e`. My best guess is some communication issue between the GUI process and the sync daemon process.

**Comment by philippgerard** [comment]
Hi Sam,

I can quote the following from the log:

```
Traceback (most recent call last):
  File "Contents/Resources/app_packages/maestral/daemon.py", line 617, in start_maestral_daemon_process
  File "Contents/Resources/app_packages/maestral/daemon.py", line 407, in wait_for_startup
  File "Contents/Resources/app_packages/maestral/daemon.py", line 403, in wait_for_startup
  File "Contents/Resources/app_packages/Pyro5/client.py", line 182, in _pyroBind
  File "Contents/Resources/app_packages/Pyro5/client.py", line 342, in __pyroCreateConnection
  File "Contents/Resources/app_packages/Pyro5/client.py", line 304, in connect_and_handshake
Pyro5.errors.CommunicationError: cannot connect to /Users/philippgerard/Library/Application Support/maestral/maestral.sock: [Errno 2] No such file or directory
2021-07-12 09:13:06 daemon ERROR: Daemon is running but not responsive, killing now
2021-07-12 09:14:58 daemon ERROR: Could not communicate with daemon
Traceback (most recent call last):
  File "Contents/Resources/app_packages/Pyro5/client.py", line 278, in connect_and_handshake
  File "Contents/Resources/app_packages/Pyro5/socketutil.py", line 283, in create_socket
FileNotFoundError: [Errno 2] No such file or directory

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "Contents/Resources/app_packages/maestral/daemon.py", line 617, in start_maestral_daemon_process
  File "Contents/Resources/app_packages/maestral/daemon.py", line 407, in wait_for_startup
  File "Contents/Resources/app_packages/maestral/daemon.py", line 403, in wait_for_startup
  File "Contents/Resources/app_packages/Pyro5/client.py", line 182, in _pyroBind
  File "Contents/Resources/app_packages/Pyro5/client.py", line 342, in __pyroCreateConnection
  File "Contents/Resources/app_packages/Pyro5/client.py", line 304, in connect_and_handshake
Pyro5.errors.CommunicationError: cannot connect to /Users/philippgerard/Library/Application Support/maestral/maestral.sock: [Errno 2] No such file or directory
2021-07-12 09:14:58 daemon ERROR: Daemon stopped with return code 0
```

Please note that I'm running on the 12.0 Monterey beta now, but the error occurred already before I updated to it, so it should not be related to Monterey (only).

**Comment by shibainu3660** [comment]
Hello! I have the same issue as well. Below please find the log. I am using M1 MacBook Air (8GB RAM, 256GB SSD) with Big Sur 11.4. Everything is fine when Maestral 1.4.4 is installed. It isn't working anymore once the version 1.4.5 is installed. The same happens after updating it to 1.4.6.

```2021-07-13 00:39:55 daemon ERROR: Could not communicate with daemon
Traceback (most recent call last):
  File "Contents/Resources/app_packages/Pyro5/client.py", line 278, in connect_and_handshake
  File "Contents/Resources/app_packages/Pyro5/socketutil.py", line 283, in create_socket
ConnectionRefusedError: [Errno 61] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "Contents/Resources/app_packages/maestral/daemon.py", line 617, in start_maestral_daemon_process
  File "Contents/Resources/app_packages/maestral/daemon.py", line 407, in wait_for_startup
  File "Contents/Resources/app_packages/maestral/daemon.py", line 403, in wait_for_startup
  File "Contents/Resources/app_packages/Pyro5/client.py", line 182, in _pyroBind
  File "Contents/Resources/app_packages/Pyro5/client.py", line 342, in __pyroCreateConnection
  File "Contents/Resources/app_packages/Pyro5/client.py", line 304, in connect_and_handshake
Pyro5.errors.CommunicationError: cannot connect to /Users/shiba/Library/Application Support/maestral/maestral.sock: [Errno 61] Connection refused
2021-07-13 00:39:55 daemon ERROR: Daemon is running but not responsive, killing now
```

**Comment by SamSchott** [comment]
Thanks for the tracebacks. Are there any preceding messages in the log before the errors? I'm looking for anything from the daemon process itself, to see how far it got when starting up. For example:

```
2021-07-12 20:25:47 daemon INFO: Starting daemon
2021-07-12 20:25:47 daemon DEBUG: Environment: {...}
2021-07-12 20:25:47 daemon DEBUG: Acquired daemon lock: ~/Library/Application Support/maestral/maestral.lock
2021-07-12 20:25:47 daemon DEBUG: Integrating with CFEventLoop
2021-07-12 20:25:47 daemon DEBUG: Socket path: ~/Library/Application Support/maestral/maestral.sock
2021-07-12 20:25:47 daemon DEBUG: Creating Pyro daemon
2021-07-12 20:25:47 daemon DEBUG: Starting event loop
```

Ideally, set the log level to DEBUG.

And just a few question to help narrow down what is going on:

1. Are you all seeing this on startup only / when "start on login" is set?
2. Do you have the daemon set to start on login independently through the CLI with `maestral autostart -Y`? 
3. Are you all using the app bundle?

Apologies for all the questions, it's just really hard to debug if cannot reproduce it.

**Comment by shibainu3660** [comment]
1. When I started to prepare the answers to the above three questions, I noticed that the log level wasn't at DEBUG. I opened up the terminal and changed the log level to DEBUG. As strange as it sounds, it opens and works fine.

Then I varied the log levels and it seems the error is reproducible on my side.
Working (i.e., Maestral works fine and syncs file as usual): DEBUG, INFO
Not Working (i.e., Maestral throws the error message "Could not access sys._trackback".): WARNING, ERROR

The way that I have tested is something like this: I opened the terminal, type "maestral log level DEBUG" and see if it opens. If so, close it and repeat the same command with other log levels.

2 and 3. Yes and I am using the app bundle. I have never used the CLI and change the settings via this mean.

**Comment by philippgerard** [comment]
Same here, log level was set to WARNING and failed to start, now with INFO the daemon starts up just fine.

`➜  ~ maestral start
Starting Maestral...        [FAILED]
Please check logs for more information.
➜  ~ maestral log show -e
➜  ~ maestral log level
Log level: WARNING
➜  ~ maestral log level INFO
✓ Log level set to INFO.
➜  ~ maestral start
Starting Maestral...        [OK]`

**Comment by SamSchott** [comment]
Yes, I can reproduce that. Apparently one of the log handlers was relying on log messages being pre-formatted by earlier handlers.

However, this seems to be a very reliable bug, fixed in the linked commit, and does not explain the flaky behaviour of sometimes failing to start.

**Comment by SamSchott** [comment]
This should be all fixed in v1.4.8.
